microsoft/testsuites/cvm/cvm_azure_host.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from pathlib import Path
+from typing import Any, Dict
+
+from assertpy.assertpy import assert_that
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+)
+from lisa.operating_system import CBLMariner
+from lisa.testsuite import TestResult
+from lisa.tools import Dmesg
+from lisa.util import LisaException, SkippedException, get_matched_str
+
+
+@TestSuiteMetadata(
+    area="cvm",
+    category="functional",
+    description="""
+    This test suite is for azure host vm pre-checks
+    for nested-cvm cases.
+    """,
+)
+class CVMAzureHostTestSuite(TestSuite):
+    __sev_enabled_pattern = re.compile(r"mshv: SEV-SNP is supported")
+    __sev_partition_pattern = re.compile(
+        r"mshv: Maximum supported SEV-SNP partitions are: (\d+)"
+    )
+
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node = kwargs["node"]
+        if not isinstance(node.os, (CBLMariner)):
+            raise SkippedException(
+                f"CVMAzureHostTestSuite is not implemented for {node.os.name}"
+            )
```

Comment:
@squirrelsc overall the code is good, per @smit-gardhariya the case will pass on mariner + DCas_cc_v5 vm sizes.
So the only concern is the test case will fail when it runs against mariner image + non DCas_cc_v5 vm size. if we can't create a feature based on vm size DCas_cc_v5, we have to judge whether the size belongs to this vm sku after vm deployment. then decide skip or continue the testing. thoughts?

Reply:
Dom0 flavor of mariner to pe precise here

Reply:
Other way can be, I can pass SKU from our testing infra as variable, and we can mark this test SKIPPED if that is None or not as per expectation (some pre-defined listed SKUs)

Reply:
@squirrelsc Could you please share your thought here.

Reply:
You can add an azure only feature, and enable this test case for specified VM sizes only.

Reply:
@squirrelsc @LiliDeng I modified before_case to add validation on VM SKU. This SKU is not publicly available yet, so i am checking it in before case instead of adding it in azure feature where we can check for SKU family.

Reply:
Please use a feature, and make vm sizes are configurable. So it can be rewritten by runbook.

Reply:
@squirrelsc Thnx for the review here. Could you please review the latest push. I have made the changes for creating feature CVMNestedVirtulization